### DISCUSSION TITLE
Can we add an agent while in step?

### DISCUSSION THREAD

**Discussion by Nagateja23** [mesa/mesa#1534] [General]
Hello Everyone,

I have a small doubt regarding ABM mesa. Like in class model we add scheduler.add_agent to add an agent, likewise is it possible to add agents while in step? Thank you in advance

**Comment by rht** [comment]
Yes, you can. Though, the list of the agents that do the `step()` is precomputed before the loop to do `step()` for each. The new agents only do `step()` during the next iteration.
